New DescriptionList, LabelValuePair components [SANDBOX]

nsmedira opened the pull request:
- new components `<DescriptionList />`, `<LabelValuePair />`
- example in `Content.jsx`
- new styling in `_lists.scss`
- component test `DescriptionList.test.js`
- add `--coverage` flag to `test` script

nsmedira committed:
install testing-library/jest-dom, react...
--coverage flag on test script

nsmedira committed:
description list component, styles and tests

nsmedira committed:
description list example

nsmedira committed:
remove unnecessary description list

nsmedira committed:
singularize author

nsmedira committed:
rename itemClass to className

nsmedira committed:
add tests for custom list, listItem classes

nsmedira committed:
handle labelValuePairs unexpected type

nsmedira committed:
remove aria-label attribute

nsmedira committed:
add comma-delimited class example

nsmedira committed:
handle null label or value...
write tests for LabelValuePair component

cwardzala commented on src/components/DescriptionList.jsx:
```diff
@@ -0,0 +1,44 @@
+    return (
+        <dl id={id || undefined} className={classnames('list-description', listClass)} data-testid="list-description">
+            {arr.map(([label, value], idx) => (
+                <LabelValuePair key={idx} label={label} value={value} className={itemClass} />
```
Keys should not be just index, maybe something like `${label.toLowerCase()}_${idx}`

cwardzala commented on src/components/DescriptionList.jsx:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    return (
+        <dl id={id || undefined} className={classnames('list-description', listClass)} data-testid="list-description">
```
Should use `@reach/auto-id` here when no `id` is provided. Look at how `form/Input.jsx` uses it.

cwardzala commented on src/components/__tests__/DescriptionList.test.js:
```diff
@@ -0,0 +1,72 @@
+});
+
+test('renders a properly shaped description list (dl)', async () => {
+    render(<DescriptionList labelValuePairs={[['label', 'value']]} />);
```
since your valid `labelValuePairs` is re-used you should just set and use a global variable.

cwardzala commented on src/components/DescriptionList.jsx:
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import classnames from 'classnames';
+
+export const LabelValuePair = ({ label = null, value = null, className = '' }) => {
```
Since this is a separate component, and its tests are in a separate file we should move this to its own file. `components/DescriptionList/LabelValuePair.jsx`.

cwardzala replied:
and think about renaming this file `components/DescriptionList/index.jsx`

cwardzala commented on src/scss/components/_lists.scss:
```diff
@@ -1,15 +1,35 @@
 ul {
     list-style-type: disc;
+
+    &.comma-delimited {
```
I would not make this a child class of `ul` and make it it's own standalone class to allow use with any element.

nsmedira committed:
apply PR feedback...
split LabelValuePair and DescriptionList into their own files
move components and __tests__ folder into components/DescriptionList
use @reach/auto-id for description list id
change DescriptionList id prop to default undefined
pass LabelValuePair a key of label text + index
new var validLabelValuePairs
alter test for absence of custom id
.comma-delimited now standalone class

nsmedira committed:
force sandbox

nsmedira committed:
add note about PR sandboxes

nsmedira committed:
add testing to GH actions

nsmedira committed:
remove --coverage flag from test script

nsmedira committed:
handle LabelValuePair keys when label is...
a react fragment
add tests for LabelValuePair keys
install react-test-renderer
